src/RRTStar/rrt_star.cpp

```diff
@@ -0,0 +1,252 @@
+#include "rrt_star.h"
+
+#include <algorithm>
+#include <utility>
+#include <vector>
+
+RRT::RRT()
+{
+    setStateSpace(START_POS_X, START_POS_Y, END_POS_X, END_POS_Y);
+    root = new Node;
+    root->parent = nullptr;
+    root->position = start_pos;
+    last_node = root;
+    nodes.push_back(root);
+    step_size = 0.5;
+    max_iter = 5000;
+}
+
+RRT::~RRT()
+{
+    reset();
+}
+
+/**
+ * @brief Initialize root node of RRT.
+ */
+void RRT::initialize()
+{
+    root = new Node;
+    root->parent = nullptr;
+    root->position = start_pos;
+    last_node = root;
+    nodes.push_back(root);
+}
+
+/**
+ * @brief Clear the RRT and free memory.
+ */
+void RRT::reset()
+{
+    for (auto obstacle : obstacles) {
+        delete obstacle;
+    }
+    obstacles.clear();
+    obstacles.resize(0);
+    deleteNodes(root);
+    nodes.clear();
+    nodes.resize(0);
+    path.clear();
+    path.resize(0);
+}
+
+void RRT::setStepSize(int step)
+{
+    step_size = step;
+}
+
+void RRT::setMaxIterations(int iter)
+{
+    max_iter = iter;
+}
+
+void RRT::setStartPosition(double x, double y) {
+    start_pos.x() = x;
+    start_pos.y() = y;
+}
+
+void RRT::setStateSpace(double x_start, double y_start, double x_end, double y_end) {
+    setStartPosition(x_start, y_start);
+    setEndPosition(x_end, y_end);
+    // State space calculated as minimum bounding rectangle with buffer
+    // origin is lower left corner, bounds is width, heigh
+    origin.x() = min(start_pos.x(), end_pos.x()) - LANE_WIDTH;
+    origin.y() = min(start_pos.y(), end_pos.y()) - LANE_WIDTH;
+    bounds.x() = max(start_pos.x(), end_pos.x()) - origin.x() + LANE_WIDTH;
+    bounds.y() = max(start_pos.y(), end_pos.y()) - origin.y() + LANE_WIDTH;
+}
+
+void RRT::setEndPosition(double x, double y) {
+    end_pos.x() = x;
+    end_pos.y() = y;
+}
+
+void RRT::addObstacle(Vector2f first_point, Vector2f second_point) {
+    obstacles.push_back(new Obstacle(std::move(first_point), std::move
+    (second_point)));
+}
+
+/**
+ * @brief Delete all nodes using DFS technique.
+ * @param root
+ */
+void RRT::deleteNodes(Node *root)
+{
+    for(auto & i : root->children) {
+        deleteNodes(i);
+    }
+    delete root;
+}
+
+/**
+ * @brief Generate a random node in the field.
+ * @return
+ */
+Node* RRT::getRandomNode()
+{
+    // Try to connect with the end occasionally
+    Node* ret;
+    if (drand48() <= 0.1) {
```

Comment:
Can you please set the seed in the constructor?

Reply:
Done

Reply:
Seeding causes consistent failure for the planning algorithm, tried with multiple random number generators. Removing for now.